packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/RestClientProvider.cs

```diff
@@ -287,10 +313,23 @@ private static IfStatement BuildQueryParameterNullCheck(
         {
             if (parameterType?.IsCollection == true)
             {
-                var changeTrackingListDeclaration = Declare(
-                    "changeTrackingList",
-                    ClientModelPlugin.Instance.TypeFactory.ListInitializationType.MakeGenericType(parameterType.Arguments),
-                    out var changeTrackingReference);
+                DeclarationExpression? changeTrackingListDeclaration;
```

Comment:
can we update this variable to be more generic now that the declaration can be either a list or dictionary ?